ktor-server/ktor-server-core/jvm/src/io/ktor/util/Parameters.kt

```diff
@@ -49,15 +53,27 @@ public inline fun Parameters.getOrFail(name: String): String {
  */
 @OptIn(ExperimentalStdlibApi::class)
 public inline fun <reified R : Any> Parameters.getOrFail(name: String): R {
-    return getOrFailImpl(name, R::class, typeOf<R>().toJavaType())
+    return getOrFailImpl(name, typeInfo<R>())
```

Comment:
`toJavaType` invocation is lost so some edge-cases may stop working because of the missing `JavaTypeAdapter`. I wonder why no tests failed because of this...

Reply:
I use default java mechanisms to get `Type`. I added tests for `WildcardType` and they pass.